config/layout.go

```diff
@@ -0,0 +1,27 @@
+package config
+
+type FilesLayout struct {
+	NumFiles          uint
+	FileNumLabels     uint64
+	LastFileNumLabels uint64
+}
+
+func DeriveFilesLayout(cfg Config, opts InitOpts) FilesLayout {
+	maxFileSizeBits := opts.MaxFileSize * 8
+	maxFileNumLabels := maxFileSizeBits / uint64(cfg.BitsPerLabel)
+	numLabels := cfg.LabelsPerUnit * uint64(opts.NumUnits)
+	numFiles := numLabels / maxFileNumLabels
```

Comment:
I think it will crash if `cfg.MaxFileSize == 0`. Perhaps add a check in `func Validate(cfg Config, opts InitOpts) error` if `cfg.MaxFileSize > 0`?

Reply:
I added `MinMaxFileSize` const and a validation for it.

Reply:
NIT: `MinFileSize` is probably a less confusing name 😅

Reply:
🤣